Ballerine/feat/kyb cf

alonp99 opened the pull request:
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

Omri-Levy committed:
refactor(workflow-node-sdk): moved createWorkflow from constructor to a public method

Omri-Levy committed:
feat(*): checkpoint

Omri-Levy committed:
feat(*): checkpoint

Omri-Levy committed:
fix(*): improved types - no longer outputs ts errors in terminal for workflow-runner

Omri-Levy committed:
fix(core and node sdk): fixed onInvokeChildWorkflow not being invoked

Omri-Levy committed:
test(*): added tests for multi state onInvokeChildWorkflow *does not pass*

Omri-Levy committed:
feat(*): child workflows now invoke `onEvent` when they reach their final state

Omri-Levy committed:
test(*): added a test to cover `onEvent` payload

Omri-Levy committed:
feat(schema.prisma): updated WorkflowDefinition to include autoEmitEvent, policy, checkPolicy

Omri-Levy committed:
refactor(*): updated contextToCopy to adhere to the same interface as api plugins

Omri-Levy committed:
feat(*): added the waitForResolved field to ChildWorkflow

Omri-Levy committed:
feat(*): added an event and action to allow updating context

Omri-Levy committed:
refactor(*): updated onEvent's payload

Omri-Levy committed:
fix(tests): now passing parent state into its metadata

Omri-Levy committed:
refactor(workflow-runner): updated onEvent to onDoneChildWorkflow and doneChildWorkflows renamed too

Omri-Levy committed:
chore(*): resolved conflicts

Omri-Levy committed:
feat(*): wIP - parent / child definitions and runtime data

Omri-Levy committed:
refactor(seed.ts): removed data from parent context

Omri-Levy committed:
refactor(*): updated child workflows api

Omri-Levy committed:
fix(package.json): removed workspace dependency

Omri-Levy committed:
fix(pnpm lockfile): re-added pnpm-lockfile.yml

Omri-Levy committed:
Child workflows API updates (#637)

* feat(*): wIP - parent / child definitions and runtime data

* refactor(seed.ts): removed data from parent context

* refactor(*): updated child workflows api

* fix(package.json): removed workspace dependency

* fix(pnpm lockfile): re-added pnpm-lockfile.yml

Omri-Levy committed:
refactor(*): no longer creating child workflow runtime data from seed

this removes runtimeId from inside workflow-runner

Omri-Levy committed:
chore(*): resolved conflicts

Omri-Levy committed:
refactor(workflows-service): removed childWorkflows method from workflow.service.ts

Blokh committed:
Blokh/feat/unified api plugin (#636)

* generate kyc plugin

* generated unifieid api logic

* feat(workflow with unified api): generated a workflow for unified api

generated the logic of getting callback

* removed unnecessary test

* fixecd ts issues

* feat: a new feature of helpers transformers

* feat(generated base64 transformation): generated base64 transformation

* fixed eslint issues

* feat(merged with feature branch): merged with feature branch

* removed unnecessary migration

* updated tranformers format

* ran format

* fixed tests

* feat(fixed tests): fixed tests

* fixed comments

Omri-Levy committed:
refactor(workflows-service): /run now handles creating parent runtime and /event for moving state

Omri-Levy committed:
refactor(workflows-service): now letting /event move the child workflow state

Omri-Levy committed:
test(workflow-node-sdk.test.ts): updated tests to match the latest changes

Omri-Levy committed:
feat(oninvokechildworkflow): added an initial sendEvent to child workflow

Omri-Levy committed:
Remove child workflow runtime data creation from seed (#640)

* feat(*): wIP - parent / child definitions and runtime data

* refactor(seed.ts): removed data from parent context

* refactor(*): updated child workflows api

* fix(package.json): removed workspace dependency

* fix(pnpm lockfile): re-added pnpm-lockfile.yml

* refactor(*): no longer creating child workflow runtime data from seed

this removes runtimeId from inside workflow-runner

* refactor(workflows-service): removed childWorkflows method from workflow.service.ts

* refactor(workflows-service): /run now handles creating parent runtime and /event for moving state

* refactor(workflows-service): now letting /event move the child workflow state

* test(workflow-node-sdk.test.ts): updated tests to match the latest changes

* feat(oninvokechildworkflow): added an initial sendEvent to child workflow

Blokh committed:
fixed issues in build

Blokh committed:
updated logic of dynamic and seed

Omri-Levy committed:
feat(*): wip - added parentContextToCopy and childContextToCopy

Omri-Levy committed:
chore(*): fixed conflicts

Omri-Levy committed:
refactor(workflow-runner): made contextToCopy optional

Omri-Levy committed:
test(workflow-node-sdk): updated test - transform is now an array

Blokh committed:
feat(fixed tests): fixed tests

Omri-Levy committed:
childContextToCopy and parentContextToCopy (#645)

* feat(*): wIP - parent / child definitions and runtime data

* refactor(seed.ts): removed data from parent context

* refactor(*): updated child workflows api

* fix(package.json): removed workspace dependency

* fix(pnpm lockfile): re-added pnpm-lockfile.yml

* refactor(*): no longer creating child workflow runtime data from seed

this removes runtimeId from inside workflow-runner

* refactor(workflows-service): removed childWorkflows method from workflow.service.ts

* refactor(workflows-service): /run now handles creating parent runtime and /event for moving state

* refactor(workflows-service): now letting /event move the child workflow state

* test(workflow-node-sdk.test.ts): updated tests to match the latest changes

* feat(oninvokechildworkflow): added an initial sendEvent to child workflow

* feat(*): wip - added parentContextToCopy and childContextToCopy

* refactor(workflow-runner): made contextToCopy optional

* test(workflow-node-sdk): updated test - transform is now an array

* feat(fixed tests): fixed tests

* refactor(types): replaced instances of 'as 'final' with 'as const'

* refactor(seed.ts): removed unused child definition

* fix(workflow-runner): pR comments changes

---------

Co-authored-by: blokh <danielblokhi@gmail.com>

alonp99 committed:
merged w/ dev

Blokh committed:
Merge branch 'omri-levy/feat/child-workflows' into omri-levy/feat/child-workflows-e-2-e

Omri-Levy committed:
feat(*): wip - can now merge multiple child workflow context into the parent

Blokh committed:
fixed entity callback request - still not working full flow

Blokh committed:
Merge branch 'omri-levy/feat/merge-child-context' into omri-levy/feat/child-workflows-e-2-e

Blokh committed:
merged conflicts

Blokh committed:
removed workflow runners logic

Blokh committed:
starting to generate aprent workflow runtime

chesterkmr committed:
feat: initialize react-ui lib & tailwind,storybook,build

Blokh committed:
feat: removd all unnecessary changes in order to completes parent workflow

chesterkmr committed:
feat: added stories & shadcn components & extended monorepo build with ui

chesterkmr committed:
fix: lock fix

Blokh committed:
finished child to parent plugin persistence logic

Blokh committed:
feat(finalized generating parent child association): finalized generating parent child association

Blokh committed:
ran format

Blokh committed:
finalized logic of iterative plugin (#663)

chesterkmr committed:
feat: added eslint & vitetest & cleaned packag.json & changed scrollbar

chesterkmr committed:
fix: replaced enum with const object

Blokh committed:
Blokh/feat/child workflow plugin (#670)

* added callback logic to child workflow

* feat(ran format + finalized workflow service callback): format + workflow service callback

chesterkmr committed:
fix: fixed build & updated test config

chesterkmr committed:
fix: updated tsconfig target & added overrides for types

chesterkmr committed:
fix: removed overrides & unused packages & context view now uses scrollarea

chesterkmr committed:
chore: merge with dev branch

chesterkmr committed:
feat: added initial kyb-example

chesterkmr committed:
feat: implemented base theming functional & added theme & settings providers

Blokh committed:
Blokh/feat/parent with child seed (#678)

* added callback logic to child workflow

* feat(ran format + finalized workflow service callback): format + workflow service callback

* generated first parent workflow

* finalized and tested callback for creation

* finished persistence of parent to child

* finalized parent persistence logic

* feat(ran format): ran format

Blokh committed:
feat(merged with dev): merged with dev

Blokh committed:
removed unnecessary import

Blokh committed:
feat: finalized parent children callback override

Blokh committed:
Merge branch 'dev' into blokh/feat/parent-workflow-proc

chesterkmr committed:
feat: added rjsv form & templates & custom fields

chesterkmr committed:
feat: added submit button layout & wf dashboard type fixes

Blokh committed:
Merge branch 'dev' into blokh/feat/parent-workflow-proc

# Conflicts:
#	pnpm-lock.yaml
#	services/workflows-service/package.json

Blokh committed:
merged with dev

chesterkmr committed:
Merge branch 'dev' into illia-rudniev/feat/kyb-example

chesterkmr committed:
feat: added rjsv templates using shadcn

Blokh committed:
generated jmespath rule engine + race condition fixing (#680)

* generated jmespath rule engine

* fixed core build

* fixing race condition

* feat: finished aggregation logic over children

* fixed eslist

* added bang for child workflow

* added bang for child workflow

* updated types

Blokh committed:
added child workflows to parent workflow format

Blokh committed:
feat: added api plugin

Blokh committed:
fixed lint

Blokh committed:
Merge branch 'blokh/feat/parent-workflow-proc' into blokh/feat/open-corp-plugin

chesterkmr committed:
feat: implemented personal info & documents forms & added types to ui package

chesterkmr committed:
Merge branch 'illia-rudniev/feat/kyb-example' into blokh/feat/open-corp-plugin

Blokh committed:
fixed ts errors

Blokh committed:
feat: added kyb plugin (#689)

* feat: added api plugin

* fixed lint

* transformed the logic and request via api

* ran format

chesterkmr committed:
feat: implemented KYC flow & controllers

chesterkmr committed:
Merge branch 'blokh/feat/open-corp-plugin' into illia-rudniev/feat/kyb-flow

chesterkmr committed:
fix: updated env example

alonp99 committed:
Merge branch 'dev' into illia-rudniev/feat/kyb-flow

alonp99 committed:
publish ui package

alonp99 committed:
publish ui package

Blokh committed:
updated migration - removed unnecessary fields

alonp99 committed:
publish ui package

alonp99 committed:
publish ui package

alonp99 committed:
publish ui package

alonp99 committed:
publish ui package

alonp99 committed:
Merge branch 'dev' into illia-rudniev/feat/kyb-flow

Omri-Levy committed:
fix(*): fixed kyb app not building due to ui lib

alonp99 committed:
publish ui package

chesterkmr committed:
fix: fixed imports in kyb & fixed label component build

chesterkmr committed:
fix: fixed cors

alonp99 committed:
publish ui package

alonp99 committed:
publish ui package

alonp99 committed:
publish ui package

alonp99 committed:
merged with dev

alonp99 committed:
chore: merge configlicts

alonp99 committed:
update packages

alonp99 committed:
update packages

alonp99 committed:
Merge branch 'blokh/feat/parent-workflow-proc' into ballerine/feat/kyb-cf

alonp99 committed:
Omri levy/refactor/global case actions (#688)

* refactor(case.actions.tsx): renamed Execute All to Ask for all revisions

* refactor(useactions): now the global case actions are enabled depending on the documents decision

* refactor(*): changed instances of resubmit to revision + no longer using dropdown

*dropdown for global case actions

* refactor(event method): the event method now handles both endUser and business

* fix(*): fixed nextEvents being always empty

* feat(*): added a revision state - ensures ask for all revisions is not always active

* feat(workflows-service): added a webhook to /event

* fix(packages/ui): renamed "lib" to "dist"

* refactor(*): moved safeEvery, someDocumentDecisionStatus, everyDocumentDecisionStatus to common

* refactor(workflows-service): created a helper to alias individual as endUser

* refactor(workflows-service): extracted shared webhooks logic to a function

* feat(workflows-service): added an entity repository which has access to both enduser and business

* fix(workflows-service): fixed dependencies for nest and tests

* revert(workflows-service): back to using configService

* fix(ci tests): fixed type of mock configService

* fix(ci): fixed tests

* fix(ci): test fixes

* fix(ci): test fixes

* fix(workflow-core): fixed failing tests

* fix(workflow-node-sdk): fixes for ci - added --passWithNoTests

* refactor(workflow-node-sdk): removed ts.ts - was committed by mistake

---------

Co-authored-by: Omri Levy <omri.levy0197@gmail.com>
Co-authored-by: Omri Levy <61207713+Omri-Levy@users.noreply.github.com>

Omri-Levy committed:
fix(package.json): changed dist/main to dist/src/main

chesterkmr committed:
feat: added demo auth guard & added env.example & added demo env

Omri-Levy commented on services/workflows-service/src/common/guards/demo.guard.ts:
```diff
@@ -0,0 +1,12 @@
+@Injectable()
+export class DemoGuard implements CanActivate {
+  canActivate(): boolean {
+    return this.isDemo() ? true : false;
```
Can be shortened to `this.isDemo()` or inline `env.NODE_ENV === 'demo'`.

alonp99 committed:
update packages

chesterkmr committed:
fix: removed demo env & added is_demo flag & updated guards & decorators & fixed kyb warnings

chesterkmr committed:
feat: added start of workflow after creation & fixed workflow-core map issue

Omri-Levy committed:
Workflows service - Prisma 400 fix (#695)

* fix(workflows-service): now passing childWorkflowsRuntimeData: true to select if select exists

* revert(workflows-service): added safeArgs to where it was missing

---------

Co-authored-by: Alon Peretz <Alonp99@gmail.com>

alonp99 committed:
collection flow data

alonp99 committed:
alonp/fix/coll data (#700)

* collection flow data

* collection flow data

* fix: formatting

---------

Co-authored-by: Illia Rudniev <cheskmr@gmail.com>

Blokh committed:
Blokh/feat/kyc email plugin (#698)

* added kyc session plugin

* generated email plugin

* finalized email plugin

* finalized email with template

* finalized email plugin

* added unified api normalizer

* ran format

Blokh committed:
ran install

Blokh committed:
finalized callback urls for plugin output using veriff

Omri-Levy committed:
fix(workflows-service): added a default case for safeArgs (select include error) (#701)

alonp99 committed:
fix(seed): merge conflicts

gst

alonp99 committed:
feat(email): email

Blokh committed:
finalized formatting hook callback handler

Blokh committed:
Merge branch 'dev' into blokh/feat/parent-workflow-proc

alonp99 committed:
collection flow data

alonp99 committed:
Merge branch 'dev' into ballerine/feat/kyb-cf

alonp99 committed:
collection flow data

Blokh committed:
updated getWorkflowByIdWithRelations to include children select

Blokh committed:
Merge branch 'blokh/feat/parent-workflow-proc' of github.com:ballerine-io/ballerine into blokh/feat/parent-workflow-proc

Blokh committed:
fixed childworkflow flow

alonp99 committed:
collection flow data

alonp99 committed:
collection flow data

chesterkmr committed:
feat: reworked workflow run payload & updated form schemas

chesterkmr committed:
fix: lock fix

chesterkmr committed:
feat: implemented workflow revision view

chesterkmr committed:
feat: implemented app rehydration & removed file storage and added base64 file parsers

Blokh committed:
updated parent complete rule

Blokh committed:
fixed rule of decistion

Blokh committed:
Merge branch 'blokh/feat/parent-workflow-proc' into ballerine/feat/kyb-cf

alonp99 committed:
Merge branch 'ballerine/feat/kyb-cf' into illia-rudniev/feat/kyb-workflow-revision

alonp99 committed:
fix(kyb-example): fix UBO correlation id

alonp99 committed:
fix(kybapp): format

gst

Blokh committed:
updated format of child workflow select

Blokh committed:
Merge branch 'blokh/feat/parent-workflow-proc' into ballerine/feat/kyb-cf

# Conflicts:
#	services/workflows-service/src/workflow/workflow.service.ts

Blokh committed:
merged with parent proc

alonp99 committed:
fix(fix): fix

Blokh committed:
fixed type issues in ubo

Blokh committed:
added country of registrationNumber to example

chesterkmr committed:
Illia rudniev/feat/UI design revision fixes (#705)

* feat: updated design & fixed fileID mapping in revision

* fix: added update workflow request & updated revision logic

* fix: fixed workflow update

Blokh committed:
fixed documents of persisted docs

Blokh committed:
Merge remote-tracking branch 'origin/ballerine/feat/kyb-cf' into ballerine/feat/kyb-cf

Blokh committed:
added document handling properly

alonp99 committed:
Merge branch 'dev' into ballerine/feat/kyb-cf

Omri-Levy committed:
Part 1/2 -  KYC block (backoffice KYB changes) (#704)

* fix(workflows-service): now passing childWorkflowsRuntimeData: true to select if select exists

* revert(workflows-service): added safeArgs to where it was missing

* feat(backoffice-v2): wip - now displaying fake child workflows (no react parts)

* refactor(backoffice-v2): clean up and made a factory for react keys

* feat(*): fixed wrong select / include config and now handling non-object address

* fix(backoffice-v2): nested component can now render falsy values (false, 0, etc.)

also disabled tailwindcss-prettier-plugin due to class names issue

* feat(nested component): nested component now handles array and object array items

* feat(*): now displaying a json dialog for array and object editable form fields

* feat(backoffice-v2): created a hard coded kyc block in storybook

* refactor(backoffice-v2): improved kyc block data

* refactor(packages/ui): removed lib (old dist) from versioning

* refactor(backoffice-v2): changed spaces for editable form

* fix(camelcasetospace): no longer turning all caps words like id to spaced letters i.e. "i d"

* feat(*): now displaying child (kyc blocks)

* feat(backoffice-v2 map): map feature now supports address as a string in addition to object

* fix(nestedcomponent): fixed label casing

* refactor(nested component): improved spacing for nested component

* refactor(nested details): improved typography hierarchy and spacing, now supporting null values

* fix(backoffice-v2): no longer preventing common keyboard shortcuts like ctrl + c

backoffice keyboard keys now require shift as well and not using event.preventDefault on unrelated
keys

* feat(backoffice-v2): nested components now support formatted dates

* feat(nested component): nested component now uses an anchor tag for valid urls with anchorUrls true

* feat(editabledetails): editableDetails now supports aliasing values, links, and labels for json

* refactor(backoffice-v2): no longer using mock workflow with children

* revert(backoffice-v2): uncommented prettier plugin as it is not the source of an issue

* add ui package to build command

---------

Co-authored-by: Alon Peretz <Alonp99@gmail.com>

alonp99 committed:
update packages

alonp99 committed:
update pnpm lock

chesterkmr committed:
fix: fixed tests

Blokh committed:
added response schema validation logic

Omri-Levy committed:
refactor(backoffice-v2): default documents empty array

Omri-Levy committed:
refactor(ui): removed lib directory

Omri-Levy committed:
perf(deployment): checking if commenting out json dialog makes deployment work

Omri-Levy committed:
perf(deployment): checking if commenting out json dialog makes deployment work (#723)

Omri-Levy committed:
test(backoffice-v2): checking if peer deps fixes issue

Omri-Levy committed:
Merge branch 'test' into ballerine/feat/kyb-cf

Omri-Levy committed:
Testing demo deployment (#724)

* perf(deployment): checking if commenting out json dialog makes deployment work

* test(backoffice-v2): checking if peer deps fixes issue

Omri-Levy committed:
fix(backoffice-v2): temp fix for json dialog not working in deployment

using a local version of json dialog in the backoffice for now

Omri-Levy committed:
Merge branch 'ballerine/feat/kyb-cf' of github.com:ballerine-io/ballerine into ballerine/feat/kyb-cf

Omri-Levy committed:
fix(backoffice-v2): temp fix for ui's json dialog - backoffice-v2 now uses local version

Omri-Levy committed:
Merge branch 'test' into ballerine/feat/kyb-cf

Blokh committed:
fixed base64 images download logic

Blokh committed:
Merge branch 'ballerine/feat/kyb-cf' of github.com:ballerine-io/ballerine into ballerine/feat/kyb-cf

alonp99 committed:
feat(local-dev): change dev cors and cookie session to allow local comm with cloud dbs

alonp99 committed:
feat(local-dev): change dev cors and cookie session to allow local comm with cloud dbs

Blokh committed:
fixed duplicated email by pre-filtering the api plugins

Blokh committed:
fixed duplicated email by pre-filtering the api plugins

Blokh committed:
Merge branch 'ballerine/feat/kyb-cf' of github.com:ballerine-io/ballerine into ballerine/feat/kyb-cf

Blokh committed:
feat(fixed pr comments): fixed pr comments

Blokh committed:
updated image-type with file-type

Blokh committed:
removed unnecessary packages

Blokh committed:
Merge branch 'dev' into ballerine/feat/kyb-cf

# Conflicts:
#	pnpm-lock.yaml
#	services/workflows-service/package.json

Blokh committed:
merged with dev

Blokh committed:
Omri levy/feat/kyb cf part 2 (#729)

* feat(*): checkpoint

* feat(backoffice-v2): now displaying a kyc block's documents (unmocked)

* feat(*): added the ability to reject, approve, and revision a kyc block

---------

Co-authored-by: Omri Levy <omri.levy0197@gmail.com>

Blokh committed:
Omri levy/feat/kyb cf part 2 (#738)

* feat(*): checkpoint

* feat(backoffice-v2): now displaying a kyc block's documents (unmocked)

* feat(*): added the ability to reject, approve, and revision a kyc block

* updated child workflow to handle re-sending email over revision

* override context document files

* Update package.json

---------

Co-authored-by: Omri Levy <omri.levy0197@gmail.com>
Co-authored-by: Alon Peretz <Alonp99@gmail.com>

alonp99 committed:
update packages

alonp99 committed:
update packages

alonp99 committed:
chore(merge-dev): merge dev to feature branch

gst

chesterkmr committed:
fix: fixed bug where uploaded files were sent without body

chesterkmr committed:
fix: lock fix

alonp99 committed:
Merge branch 'dev' into ballerine/feat/kyb-cf

alonp99 committed:
tests fix

Blokh committed:
Blokh/feat/kyb resubmit (#739)

* updated kyb flow logic

* finalized parent workflow definition

* added kyc resubmission template

* finalized kyc flow with resubmission

---------

Co-authored-by: Alon Peretz <Alonp99@gmail.com>

chesterkmr committed:
Illia rudniev/feat/kyb stepper (#735)

* feat: added UI stepper implementation & added headless hook to work with Stepper

* feat: implemented KYB flow using stepper

* feat: implemented data persistance with stepper & refactored code

* feat: implemented rendering of revision steps & updated serializers & add Issue Resolving flow

* fix: updated detection of completed steps from context

* feat: added global workflow state so data could be resotred on revision flow

* feat: implemented warnings render & reworked interaction with storage

* fix: fixed flow data updating & mapping & added rendering of warnings on documents view

* fix: fixed mapping & status resolving bugs & added missing final screen

* feat: added partial workflow update(documents)

* chore: removed unused libs

* feat: added country selection input & added command and popover components to UI

* feat: up version bump & lock fix

* update packages

* fix: lock fix

* fix: updated UI dependencies & version bump to fix build issue

* update packages

* update packages

---------

Co-authored-by: Alon Peretz <Alonp99@gmail.com>

alonp99 committed:
tests fix

Omri-Levy committed:
Blocks API (as a package) (#722)

* feat(*): added a package for the blocks api

* fix(package.json): added --passWithNoTests to let CI pass

* fix(types): can now see all added blocks and cells on the type level

* feat(blocks): added a blocks component, fixed types for first cell of first block

* fix(blocks): fixed block type and renamed blocks component to avoid conflict with interface

* feat(blocks): improved type safety on the Block property of the BlocksComponent

* test(blocks): added spec tests to the blocks package

* feat(blocks): added debugging means to the blocks api

* feat(blocks): added a way to access blocks, blocks count, and cells count

* feat(blocks): added the ability to start with initial blocks

* feat(blocks): added the ability to merge two block arrays

* feat(blocks): added storybook to help develop the blocks component

* refactor(blocks): added the ability to add blocks and pass initial blocks - wip type tests

currently the types for `addBlocks` and `initialBlocks` are broken`

* feat(blocks): added type tests, finalized first draft of api and features

* refactor(version control): removed storybook-static from version control

* refactor(version control): removed changes which were supposed to be unversioned

* refactor(*): removed changes from bad conflict resolution

* merged with feature branch

---------

Co-authored-by: Alon Peretz <Alonp99@gmail.com>

Blokh committed:
updated decision placement

chesterkmr committed:
fix: fixed base64 image converts fromBase64 & toFile

alonp99 committed:
update licsenses

alonp99 committed:
update licsenses

chesterkmr committed:
fix: fixed ui build issue & version bump (#749)

* fix: fixed ui build issue & version bump

* fix: lock fix

---------

Co-authored-by: Alon Peretz <Alonp99@gmail.com>

alonp99 committed:
update packages

alonp99 committed:
chore(new versions): new version

Blokh committed:
Merge branch 'dev' into ballerine/feat/kyb-cf

Omri-Levy committed:
fix(version control): fixed conflicts between dev and ballerine/feat/kyc-cf
